libcaf_core/src/detail/type_id_list_builder.cpp

```diff
@@ -135,14 +139,15 @@ type_id_list type_id_list_builder::move_to_list() {
 }
 
 type_id_list type_id_list_builder::copy_to_list() const {
-  if (size_ == 0)
+  auto list_size = size();
+  if (list_size == 0)
     return make_type_id_list();
-  storage_[0] = static_cast<type_id_t>(size());
   auto vptr = malloc(size_ * sizeof(type_id_t));
   if (vptr == nullptr)
     throw std::bad_alloc();
   auto copy = reinterpret_cast<type_id_t*>(vptr);
-  memcpy(copy, storage_, size_ * sizeof(type_id_t));
+  copy[0] = static_cast<type_id_t>(list_size);
+  memcpy(copy + 1, storage_ + 1, list_size * sizeof(type_id_t));
```

Comment:
Why did you change this?

Reply:
Because `storage_[0]` doesn't have the final size at this point.

Reply:
Ok. That is what line 140 ensured before and since you removed that you now have to set the size of the copy explicitly?

Reply:
Line 140 before was flat out wrong. A `const` member function should not modify the content of the object. 😉

Reply:
🤭